### ISSUE TITLE
Navmesh Surface doesn't bake terrain trees

### ISSUE THREAD

**Issue by Arugin** [Unity-Technologies/NavMeshComponents#30]
For both options "Use Geometry".

Trees has colliders and "Enable Tree Collider" option in the Terrain Collider is checked.

**Comment by rhinboost** [comment]
I am experiencing the same issue using TC2, no matter what kind of geometry I use, it does not bake it around the trees. If I drag & drop the tree prefab and rebake, it actually works, but that is not a solution for me.

**Comment by wes-hawkins** [comment]
This is a huge oversight...

**Comment by maikramer** [comment]
I have the same issue, i tried everything, using the old editor navigation baker, works in the same terrain with the same tree, then, has to be something related to navmeshcomponents, this is a blocking issue to me.

**Comment by turadr** [comment]
Yes, this is a known issue. Unfortunately it hasn't been addressed yet.

**Comment by NaaleiTB** [comment]
Hope it can help. This is not a fully working solution because I have not handled all cases but I think it can easily be tweaked to your convenience.

https://forum.unity.com/threads/does-the-component-navigation-work-with-terrain-trees-yet.488074/#post-3355183

**Comment by Bjakuja** [comment]
Still cannot find some workaround with this, trees are not baked if they are painted to terrain, the system just ignores them. The solution above not working for me, like components and would to use them but have these problems. Anyone help? Thanks

**Comment by OndrejPetrzilka** [comment]
I can't believe somebody just forgot about trees, huge oversight.

**Comment by angi-** [comment]
Has this issue been addressed yet?

**Comment by 2kCarlos** [comment]
I just left a comment (as ModLunar) here on the forums with an adapted solution, thanks to someone's initial help: https://forum.unity.com/threads/does-the-component-navigation-work-with-terrain-trees-yet.488074/#post-4929011

This adds support for trees with NavMeshObstacles on their prefab's root GameObject. Hope this helps, cause it sure saved my game :)

**Comment by uladzislaubasin** [comment]
are there any plans to address this bug in near future?

**Comment by turadr** [comment]
We have the issue logged in our bug tracking system but I can't estimate any date when it will be investigated and solved.

**Comment by angi-** [comment]
> We have the issue logged in our bug tracking system but I can't estimate any date when it will be investigated and solved. Most probably not in the near future.

I wrote about this 2 years ago. Sigh...

**Comment by OneManMonkeySquad** [comment]
Why is this so hard? In the linked thread there's a modified NavMeshSurface script floating around that takes terrain trees into account... Any reason this is not adopted into this repo?

**Comment by turadr** [comment]
It needs more testing in order to make sure that some rare situations are covered as well.
